### PR TITLE
Bulkblocks and Files tests

### PR DESCRIPTION
**Pull request by d-ylee**
Added Bulkblocks POST tests for parent and child.

Implemented Files GET tests that contain block_name.

### CODE REVIEW

**Review by vkuznet — COMMENTED** [review]
Dennis, I'll review your PR tomorrow, but so far the only question I have is that this PR seems only address `bulkblocks.go` while we have two implementations. The only you're using here and another is concurrent implementation which you can find in `bulkblocks2.go`. The DBS server can be configured to use either one or another via `abd.ConcurrentBulkBlocks` parameter. I would expect that this PR should cover both use-case. Please let me know if you consider doing that.

**Comment by d-ylee** [comment]
@vkuznet I initially did the Bulkblocks insert in order to write the files GET tests with the ranged run_num parameter. I can add on a ConcurrentBulkBlocks insert test.

File: dbs/bulkblocks.go [hunk 1]
```diff
@@ -583,6 +583,15 @@ func (a *API) InsertBulkBlocks() error {
 		if lBy == "" {
 			lBy = a.CreateBy
 		}
+		// if the data string does not contain the is_file_valid field
```
Comment (vkuznet) [inline dbs/bulkblocks.go:586]: I think we need identical block of code in `bulkblocks2.go` too

File: test/integration_test.go [hunk 2]
```diff
@@ -36,10 +36,12 @@ func runTestWorkflow(t *testing.T, c EndpointTestCase) {
 			t.Run(v.description, func(t *testing.T) {
 
 				// set the default handler
-				handler := c.defaultHandler
-				if v.handler != nil {
-					handler = v.handler
-				}
+				/*
```
Comment (vkuznet) [inline test/integration_test.go:39]: if this block of code is no longer required could you please remove it

File: test/integration_test.go [hunk 3]
```diff
@@ -85,20 +87,23 @@ func runTestWorkflow(t *testing.T, c EndpointTestCase) {
...
 					}
+
 					verifyResponse(t, d, v.output)
 				}
-				if v.method == "POST" {
-					rURL := parseURL(t, server.URL, endpoint, v.params)
-					rr, err := respRecorder("GET", rURL.RequestURI(), nil, handler)
-					if err != nil {
-						t.Error(err)
+				/*
```
Comment (vkuznet) [inline test/integration_test.go:93]: I'm not sure why you remove POST method check, could you please explain this

**Review by vkuznet — CHANGES_REQUESTED** [review]
I suggest to wait on this PR merge when you'll provide similar integration tests for concurrent bulk block insert. I also put few minor comments which I want to clarify with you before the merge.

**Review by vkuznet — CHANGES_REQUESTED** [review]
Dennis, I think having  separate `EndpointTestCase` in `getConcurrentBulkBlocksTestTable` and `getBulkBlocksTestTable ` functions is overkill since they suppose to do the same thing. Instead I suggest to define a single helper function, e.g. `testCaseBulkBlocks` where you can pass concurrent argument and it will create proper `EndpointTestCase` structure. Then, you may use this structure within `getConcurrentBulkBlocksTestTable` and `getBulkBlocksTestTable ` functions . For instance, 
```
func testCaseBulkBlocks(t *testing.T, concurrent bool) EndpointTestCase {
      blockName := []string{TestData.ParentStepchainBlock}
      if concurrent {
           blockName := []string{TestData.ParentStepchainBlock+"2"}
      }
      return EndpointTestCase{
           ....
           concurrentBulkBlocks: concurrent
            ....
                params : urlValues{"block_name": blockName}
      }
}
```
and, then you'll simplify `getConcurrentBulkBlocksTestTable` and `getBulkBlocksTestTable ` functions to
```
func getConcurrentBulkBlocksTestTable(t *testing.T) EndpointTestCase {
     return testCaseBulkBlocks(t, true)
}
func getBulkBlocksTestTable(t *testing.T) EndpointTestCase {
      return testCaseBulkBlocks(t, false)
}
```

**Review by vkuznet — APPROVED** [review]

**Pull request merged**